Mathlib/Order/WellFoundedSet.lean

```diff
@@ -234,69 +245,71 @@ theorem isWF_iff_no_descending_seq :
 
 end Preorder
 
-/-!
-### Partially well-ordered sets
-
-A set is partially well-ordered by a relation `r` when any infinite sequence contains two elements
-where the first is related to the second by `r`. Equivalently, any antichain (see `IsAntichain`) is
-finite, see `Set.partiallyWellOrderedOn_iff_finite_antichains`.
--/
+/-! ### Partially well-ordered sets -/
 
+/-- `s.PartiallyWellOrderedOn r` indicates that the relation `r` is `WellQuasiOrdered` when
+restricted to `s`.
 
-/-- A subset is partially well-ordered by a relation `r` when any infinite sequence contains
-  two elements where the first is related to the second by `r`. -/
+TODO: rename this to `WellQuasiOrderedOn` to match `WellQuasiOrdered`. -/
 def PartiallyWellOrderedOn (s : Set α) (r : α → α → Prop) : Prop :=
-  ∀ f : ℕ → α, (∀ n, f n ∈ s) → ∃ m n : ℕ, m < n ∧ r (f m) (f n)
+  WellQuasiOrdered (Subrel r (· ∈ s))
 
 section PartiallyWellOrderedOn
 
 variable {r : α → α → Prop} {r' : β → β → Prop} {f : α → β} {s : Set α} {t : Set α} {a : α}
 
+theorem PartiallyWellOrderedOn.exists_lt (hs : s.PartiallyWellOrderedOn r) {f : ℕ → α}
+    (hf : ∀ n, f n ∈ s) : ∃ m n, m < n ∧ r (f m) (f n) :=
+  hs fun n ↦ ⟨_, hf n⟩
+
+theorem partiallyWellOrderedOn_iff_exists_lt : s.PartiallyWellOrderedOn r ↔
+    ∀ f : ℕ → α, (∀ n, f n ∈ s) → ∃ m n, m < n ∧ r (f m) (f n) :=
+  ⟨PartiallyWellOrderedOn.exists_lt, fun hf f ↦ hf _ fun n ↦ (f n).2⟩
+
 theorem PartiallyWellOrderedOn.mono (ht : t.PartiallyWellOrderedOn r) (h : s ⊆ t) :
-    s.PartiallyWellOrderedOn r := fun f hf => ht f fun n => h <| hf n
+    s.PartiallyWellOrderedOn r :=
+  fun f ↦ ht (Set.inclusion h ∘ f)
 
 @[simp]
-theorem partiallyWellOrderedOn_empty (r : α → α → Prop) : PartiallyWellOrderedOn ∅ r := fun _ h =>
-  (h 0).elim
+theorem partiallyWellOrderedOn_empty (r : α → α → Prop) : PartiallyWellOrderedOn ∅ r :=
+  wellQuasiOrdered_of_isEmpty _
 
 theorem PartiallyWellOrderedOn.union (hs : s.PartiallyWellOrderedOn r)
     (ht : t.PartiallyWellOrderedOn r) : (s ∪ t).PartiallyWellOrderedOn r := by
-  rintro f hf
-  rcases Nat.exists_subseq_of_forall_mem_union f hf with ⟨g, hgs | hgt⟩
-  · rcases hs _ hgs with ⟨m, n, hlt, hr⟩
+  intro f
+  obtain ⟨g, hgs | hgt⟩ := Nat.exists_subseq_of_forall_mem_union _ fun x ↦ (f x).2
+  · rcases hs.exists_lt hgs with ⟨m, n, hlt, hr⟩
     exact ⟨g m, g n, g.strictMono hlt, hr⟩
-  · rcases ht _ hgt with ⟨m, n, hlt, hr⟩
+  · rcases ht.exists_lt hgt with ⟨m, n, hlt, hr⟩
     exact ⟨g m, g n, g.strictMono hlt, hr⟩
 
 @[simp]
 theorem partiallyWellOrderedOn_union :
     (s ∪ t).PartiallyWellOrderedOn r ↔ s.PartiallyWellOrderedOn r ∧ t.PartiallyWellOrderedOn r :=
-  ⟨fun h => ⟨h.mono subset_union_left, h.mono subset_union_right⟩, fun h =>
-    h.1.union h.2⟩
+  ⟨fun h ↦ ⟨h.mono subset_union_left, h.mono subset_union_right⟩, fun h ↦ h.1.union h.2⟩
```

Comment:
In this case it's fine, but in the future please avoid making stylistic changes in a PR that isn't otherwise about style, particularly so in a refactor PR.